unity-renderer/Assets/DCLPlugins/WorldsPlugin/Tests/WorldsPluginTests.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using DCL;
+using DCL.Controllers;
+using Decentraland.Bff;
+using NSubstitute;
+using NUnit.Framework;
+using UnityEngine;
+using Variables.RealmsInfo;
+
+public class WorldsPluginTests
+{
+
+    private WorldsPlugin worldsPlugin;
+    private string catalystRealmName = "CatalystRealmName";
+    private string worldRealmName = "WorldRealmName";
+    private IWorldsModifier genericModifier;
+    private WorldsBlockerModifier worldsBlockerModiferSubstitute;
+    private ServiceLocator serviceLocator;
+    
+    [SetUp]
+    public void SetUp()
+    {
+        // This is need to sue the TeleportController
+        serviceLocator = ServiceLocatorTestFactory.CreateMocked();
+        Environment.Setup(serviceLocator);
+        
+        worldsPlugin = new WorldsPlugin();
+        worldsBlockerModiferSubstitute = Substitute.For<WorldsBlockerModifier>();
+        genericModifier = Substitute.For<IWorldsModifier>();
+        List<IWorldsModifier> substituteModifiers = new List<IWorldsModifier>() { worldsBlockerModiferSubstitute, genericModifier };
+        worldsPlugin.worldsModifiers = substituteModifiers;
+        
+        SetCatalystRealmsInfo();
+    }
+    
+    [Test]
+    public void ModifierCalledOnRealmChange()
+    {
+        SetCatalystRealm();
+        genericModifier.Received().EnteredRealm(true, Arg.Any<AboutResponse >());
+
+        SetWorldsRealm();
+        genericModifier.Received().EnteredRealm(false, Arg.Any<AboutResponse >());
+    }
+
+    [Test]
+    public void GreenBlockerAddedOnRealmChange()
+    {
+        SetCatalystRealm();
+        serviceLocator.Get<IWorldBlockersController>().Received().SetEnabled(true);
+
+        SetWorldsRealm();
+        serviceLocator.Get<IWorldBlockersController>().Received().SetEnabled(false);
+    }
+
+    private void SetCatalystRealm()
+    {
+        DataStore.i.realm.playerRealmAbout.Set(new AboutResponse
+        {
+            Bff = new AboutResponse.Types.BffInfo(),
+            Comms = new AboutResponse.Types.CommsInfo(),
+            Configurations = new AboutResponse.Types.AboutConfiguration()
+            {
+                RealmName = catalystRealmName
+            }
+        });
+    }
+    
+    private void SetWorldsRealm()
+    {
+        DataStore.i.realm.playerRealmAbout.Set(new AboutResponse
+        {
+            Bff = new AboutResponse.Types.BffInfo(),
+            Comms = new AboutResponse.Types.CommsInfo(),
+            Configurations = new AboutResponse.Types.AboutConfiguration()
+            {
+                RealmName = worldRealmName
+            }
+        });
+    }
```

Comment:
this is copy-past thing (remember DRY principle ;) ) that can be a good improvement for the test. 

1. refactor these 2 functions into one `function SetRealm(string realmName);` 
2. refactor first Test with TestCases
```
[TestCase(catalystRealmName, true)]
    [TestCase(worldRealmName, false)]
    public void ModifierCalledOnRealmChange(string realmName, bool isCatalist)
    {
        // Act
        SetRealm(realmName);
        
        // Assert
        genericModifier.Received().EnteredRealm(isCatalist, Arg.Any<AboutResponse >());
    }
```

Reply:
3. you can also re-write another Test then. I used TestCases, but probably could be arranged more beautifull with TestCaseSource approach
```
[TestCase(new [] { catalystRealmName, worldRealmName }, new [] { true, false })]
    [TestCase(new [] { worldRealmName, catalystRealmName }, new [] { false, true })]
    [TestCase(new [] { worldRealmName, catalystRealmName, worldRealmName }, new [] { false, true, false })]
    public void GreenBlockerAddedOnRealmChange(string[] realmNames, bool[] isGreenBlockerEnabled)
    {
        for (int i = 0; i < realmNames.Length; i++)
        {
            SetRealm(realmNames[i]);
            serviceLocator.Get<IWorldBlockersController>().Received().SetEnabled(isGreenBlockerEnabled[i]);
        }
    }
```